File: lib/json_api/webhook.rb
module JsonApi::Webhook
  extend JsonApi::Json
  
  TYPE_KEY = 'webhook'
  DEFAULT_PAGE = 10
  MAX_PAGE = 25
    
  def self.build_json(obj, args={})
    json = {}
    # TODO: include devices
    
    json['id'] = obj.global_id
    json['name'] = obj.settings['name'] || 'Unnamed Webhook'
    json['url'] = obj.settings['url'] if obj.settings['url']
    json['webhook_type'] = obj.settings['webhook_type']
    json['webhooks'] = (obj.settings['notifications'] || {}).keys
    includes_content = false
    (obj.settings['notifications'] || {}).each do |key, list|
      list.each do |notif|
        includes_content = true if notif['include_content']
      end
    end
    json['include_content'] = !!includes_content
    json['advanced_configuration'] = !!obj.settings['advanced_configuration']
    json['custom_configuration'] = !!obj.settings['url']

    json
  end
end

File: app/frontend/app/models/webhook.js
import Ember from 'ember';
import DS from 'ember-data';
import CoughDrop from '../app';
import persistence from '../utils/persistence';

CoughDrop.Webhook = DS.Model.extend({
  name: DS.attr('string'),
  user_id: DS.attr('string'),
  url: DS.attr('string'),
  webhook_type: DS.attr('string'),
  webhooks: DS.attr('raw'),
  notifications: DS.attr('raw'),
  include_content: DS.attr('boolean'),
  content_type: DS.attr('raw'),
  advanced_configuration: DS.attr('boolean'),
  custom_configuration: DS.attr('boolean'),
  webhooks_list: function() {
    return (this.get('webhooks') || []).join(', ');
  }.property('webhooks')
});

export default CoughDrop.Webhook;
